packages/story-editor/src/components/toggleButton/index.js

```diff
@@ -35,57 +39,135 @@ import { forwardRef } from '@googleforcreators/react';
+          preset: {
+            ...theme.typography.presets.paragraph[
+              THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+            ],
+          },
+          theme,
+        })};
+
+        .badge {
+          margin: 0 8px;
+        }
+      `
+    }
+
+    .badge.open,
+    &:hover .badge,
+    &:focus .badge {
+      background-color: ${theme.colors.bg.quaternary};
+    }
+
+  }`}
+
+  .badge {
+    transition: background-color ${BUTTON_TRANSITION_TIMING};
+  }
+
+  // Margin is set to -8px to compensate for empty space
+  // around the actual icon graphic in the svg
+  svg.main-icon {
+    height: 32px;
+    width: auto;
+    margin: -8px;
+    display: block;
+  }
 `;
 Button.propTypes = {
   hasText: PropTypes.bool,
   isOpen: PropTypes.bool,
+  $size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
 };
 
-const Wrapper = styled.div`
-  width: 100%;
-  height: 100%;
-  display: flex;
-  justify-content: space-around;
-  align-items: center;
-`;
+// `badgeSize` is the default height and min-width of the badge circle/oval
+// Hard-coded here, but can be overridden by `$size` prop in the component
+const badgeSize = 22;
+
+// Defaults for badge come from typography.presets.label.small
+const badgeText =
+  dsTheme.typography.presets.label[
+    THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+  ];
 
-const NotificationCount = styled(Text).attrs({ as: 'span' })`
-  margin: -2px 0 -2px;
-  padding-right: 6px;
-  line-height: 20px;
-  color: ${({ theme }) => theme.colors.violet[20]};
+// TODO: Extract `CountBadge` to its own component
+// Props prefixed with `$` won't leak into element attributes
+// (`theme` is handled internally by Styled Components)
+// padding and border-radius properties use css `calc()` so units
+// other than px can be used for $size and still display correctly
+const CountBadge = styled(Text).attrs({ as: 'span', className: 'badge' })`
+  ${({ $size = badgeSize, $fontSize = badgeText.size, theme }) => css`
+    min-width: ${resolveSizeUnits($size)};
+    width: auto;
+    height: ${resolveSizeUnits($size)};
+    padding: 0 calc(${resolveSizeUnits($size)} / 4);
+    margin-left: 10px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    color: ${theme.colors.fg.primary};
+    background-color: ${theme.colors.bg.tertiary};
+    border-radius: calc(${resolveSizeUnits($size)} / 2);
+    font-size: ${resolveSizeUnits($fontSize)};
+    line-height: 0;
+  `}
 `;
+CountBadge.propTypes = {
+  $size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  $fontSize: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const DisclosureIcon = styled(Icons.ChevronDownSmall)(
+  ({ theme }) => css`
+    height: 32px;
+    width: auto;
+    margin: 0 -10px;
+    color: ${theme.colors.fg.secondary};
+    transition: transform 100ms;
+
+    &.open {
+      transform: rotate(180deg);
+    }
+  `
+);
+
+// TODO: Extract 'Disclosure' to its own component
```

Comment:
Yes please, this appears to be duplicated code.

Reply:
Done

Reply:
@barklund In extracting this component, I applied negative margins to the icon `svg` because the actual graphic has a lot of empty space around it. For regular icons, it makes sense to keep the margin as-is, but I feel like spacing of a little disclosure chevron should only occupy as much space as the graphic itself.

The menus on the dashboard were affected by the styling change, but IMO, the padding should be applied to the menu, not the contents.

<img width="623" alt="image" src="https://user-images.githubusercontent.com/4173034/174842883-6584e2ac-b11d-4030-bf59-0a43e8e6d07c.png">

Thoughts?